storage/client/queue/src/main/java/com/azure/storage/queue/QueueServiceClientBuilder.java

```diff
@@ -230,11 +233,28 @@ public QueueServiceClientBuilder credential(SharedKeyCredential credential) {
      * @throws NullPointerException If {@code connectionString} is {@code null}.
      */
     public QueueServiceClientBuilder connectionString(String connectionString) {
-        Objects.requireNonNull(connectionString);
         this.sharedKeyCredential = SharedKeyCredential.fromConnectionString(connectionString);
+        getEndPointFromConnectionString(connectionString);
```

Comment:
Don't see any javadoc changes
Should they be updated in the builder class that endpoint is optional?

Reply:
It is hard to state optional as it is optional only if connectionString built in. If we only build in credential, then endpoint is still necessary.

Reply:
okay, so are these scenarios specified in javadocs?
It will be helpful for the user I feel.

Reply:
There are descriptions in builder class. I can add some additional doc to make it more clear.